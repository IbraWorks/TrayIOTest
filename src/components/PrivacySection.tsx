import React from "react";
import { IPrivacyValues } from "../interfaces/IFormValues";
import { Form, Button } from "semantic-ui-react";
import { useForm } from "react-hook-form";

interface IProps {
  handleData: (data: IPrivacyValues) => void;
  privacyValues: IPrivacyValues;
}

const PrivacySection: React.FC<IProps> = ({ privacyValues, handleData }) => {
  const { register, handleSubmit } = useForm();

  const onSubmit = (data: any) => {
		handleData(data);
  };

  return (
    <Form onSubmit={handleSubmit(onSubmit)}>
      <Form.Field>
        <label>
          Receive updates about Tray.io product by email
          <input
            style={{ float: "right", marginTop: "5px" }}
            name="updatesByEmail"
            type="checkbox"
            placeholder="Software Engineer"
            defaultValue={privacyValues.updatesByEmail}
            ref={register()}
          />
        </label>
      </Form.Field>
      <Form.Field>
        <label>
          Receive updates about Tray.io product by email
          <input
            style={{ float: "right", marginTop: "5px"}}
            name="communicationByEmail"
            type="checkbox"
            placeholder="Software Engineer"
            defaultValue={privacyValues.communicationByEmail}
            ref={register()}
          />
        </label>
      </Form.Field>
      <Button color="green" style={{ float: "right" }} type="submit">
        Submit
      </Button>
    </Form>
  );
};

export default PrivacySection;
