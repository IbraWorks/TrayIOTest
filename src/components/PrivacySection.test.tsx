import { shallow } from 'enzyme'
import PrivacySection from './PrivacySection'
import React from 'react'

describe('PrivacySection', () => {
    it('should render', () => {
        const mockHandleData = jest.fn();
        const privacyValues = {
            updatesByEmail: "",
            communicationByEmail: ""
        }

        const wrapper = shallow(<PrivacySection handleData={mockHandleData} privacyValues={privacyValues}/>)
        expect(wrapper.exists()).toBe(true)
    });

    it('matches snapshot', () => {

        const mockHandleData = jest.fn();
        const privacyValues = {
            updatesByEmail: "",
            communicationByEmail: ""
        }

        const wrapper = shallow(<PrivacySection handleData={mockHandleData} privacyValues={privacyValues}/>);
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('submitting the form should call handle data', () => {
        const mockHandleData = jest.fn();
        const privacyValues = {
            updatesByEmail: "",
            communicationByEmail: ""
        }

        const wrapper = shallow(<PrivacySection handleData={mockHandleData} privacyValues={privacyValues}/>)
        wrapper.find('Form').simulate('submit')
        expect(mockHandleData).toBeCalledTimes(1)
    });

    it('has updatesByEmail input', () => {

        const mockHandleData = jest.fn();
        const privacyValues = {
            updatesByEmail: "",
            communicationByEmail: ""
        }

        const wrapper = shallow(<PrivacySection handleData={mockHandleData} privacyValues={privacyValues}/>);
        expect(wrapper.find("input[name='updatesByEmail']").length).toEqual(1);
    });

    it('has communicationByEmail input', () => {

        const mockHandleData = jest.fn();
        const privacyValues = {
            updatesByEmail: "",
            communicationByEmail: ""
        }

        const wrapper = shallow(<PrivacySection handleData={mockHandleData} privacyValues={privacyValues}/>);
        expect(wrapper.find("input[name='communicationByEmail']").length).toEqual(1);
    });

    it('has submit button', () => {

        const mockHandleData = jest.fn();
        const privacyValues = {
            updatesByEmail: "",
            communicationByEmail: ""
        }

        const wrapper = shallow(<PrivacySection handleData={mockHandleData} privacyValues={privacyValues}/>);
        expect(wrapper.find("Button").length).toEqual(1);
        expect(wrapper.find('Button').props()).toEqual({
            "as": "button",
            "children": "Submit",
            "color": "green",
            "style": {float: "right"},
            "type": "submit",
        })
    });

    
})
