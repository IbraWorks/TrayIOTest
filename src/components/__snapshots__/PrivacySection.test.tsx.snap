// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivacySection matches snapshot 1`] = `"<form class=\\"ui form\\"><div class=\\"field\\"><label>Receive updates about Tray.io product by email<input type=\\"checkbox\\" style=\\"float:right;margin-top:5px\\" name=\\"updatesByEmail\\" placeholder=\\"Software Engineer\\" value=\\"\\"/></label></div><div class=\\"field\\"><label>Receive communication by email for other products created by the Tray.io team<input type=\\"checkbox\\" style=\\"float:right;margin-top:5px\\" name=\\"communicationByEmail\\" placeholder=\\"Software Engineer\\" value=\\"\\"/></label></div><button style=\\"float:right\\" type=\\"submit\\" class=\\"ui green button\\">Submit</button></form>"`;
